### PR TITLE
[ADD] Add module project_assignment

### PR DESCRIPTION
**Pull request by YannickB**
This module allow to automatically assign user to the task when we change the stage of the issue. We can specify the partner on the stage, and overwrite the configuration in project and in the task.

This module use base_recursive_model for the configuration inheritance, currently submitted in server-tools. Check this PR : https://github.com/OCA/server-tools/pull/94

### CODE REVIEW

File: project_assignment/project_assignment.py [hunk 1]
```diff
@@ -0,0 +1,263 @@
+# -*- coding: utf-8 -*-
```
Comment (bwrsandman) [inline project_assignment/project_assignment.py]: Please split this file by model/class

File: project_assignment/project_assignment_view.xml [hunk 2]
```diff
@@ -0,0 +1,135 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (bwrsandman) [inline project_assignment/project_assignment_view.xml]: Please split this file per model.
- project_assigned_partner_config_view.xml
- project_project_view.xml
- project_task_view.xml
- project_task_type_view.xml

File: project_assignment/data/project_assignment_demo.xml [hunk 3]
```diff
@@ -0,0 +1,52 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (bwrsandman) [inline project_assignment/data/project_assignment_demo.xml]: Please split this file per model:
- project_task_type_demo.xml
- project_project_demo.xml
- project_task_demo.xml

File: project_assignment/README.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+- * INHERIT project.project.tree.assignpartner (tree)
+- * INHERIT project.task.form.assignpartner (form)
+- * INHERIT project.task.search.assignpartner (search)
+- * INHERIT project.task.tree.assignpartner (tree)
+- * INHERIT project.task.type.form.assignpartner (form)
+- project.assigned.partner.config.form (form)
+- project.assigned.partner.config.tree (tree)
+
+## Dependencies ##
+- base_recursive_model	vc
```
Comment (pedrobaeza) [inline project_assignment/README.md]: You have to indicate that the module is hosted (hopefully) on server-tools repo (and its URL).

File: project_assignment/README.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
+# Project Assignment #
+**Automatically reassign task to specified partner depending on stage**
+
+## Functions ##
+- For each stage, the partner can be specified in stage, then in project and finally task itself
+- We use partner instead of user for more flexibility
+- Use base inherit config
+
+## Created Views ##
```
Comment (pedrobaeza) [inline project_assignment/README.md]: It is not needed such detail. This is also given by the module technical data tab.

File: project_assignment/__init__.py [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+import project_assignment
```
Comment (pedrobaeza) [inline project_assignment/__init__.py]: `from .`

File: project_assignment/__openerp__.py [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+##############################################################################
+
+{
+    'name': 'Project Assignment',
+    'version': '1.0',
+    'category': 'Project',
+    'author': 'Yannick Buron',
+    'license': 'AGPL-3',
+    'description': """
+Project Assignment
```
Comment (pedrobaeza) [inline project_assignment/__openerp__.py]: Remove description key, as a README file is already present. You should also put it as README.rst to be correctly parsed by both Github and Odoo.

File: project_assignment/data/project_assignment_demo.xml (hunk 3 above)
Comment (YannickB) [inline project_assignment/data/project_assignment_demo.xml]: ok for project_task_type but I need project and task to be in one file. When we trigger the write in project_demo the task must already be created so it's assignment will be updated.

**Comment by YannickB** [comment]
All corrections shall be done, now I am waiting for base_recursive_model to be merge in server-tools

**Comment by dreispt** [comment]
Set to "Work in progress" since dependency is not yet available.

**Comment by rafaelbn** [comment]
This PR is from 2014. What do you think @YannickB ?

@dreispt we should close it

**Comment by dreispt** [comment]
I am using this feature, and implemented it using Automated Actions.
Do you find limitations there preventing from doing what you need?

**Comment by rafaelbn** [comment]
@dreispt I think this functionality is interesting but it look PR is forgotten. So to have OCA clear we must close or finish all PR's from 2014 at least. What do you want to do?

I'm not sure if @YannickB is going to answer.

**Comment by YannickB** [comment]
Hi @rafaelbn , unfortunately I think we should close since this merge required another PR on another branch and we're lacking time to do it. I'll reopen the PR if the need rise again, the code is not loss.

**Comment by dreispt** [comment]
@YannickB closing it as per your comment.
Feel welcome to come back at it with you can.

**Pull request closed without merging**